packages/fdc3-agent-proxy/test/step-definitions/channels.steps.ts

```diff
@@ -180,14 +178,17 @@ Given('{string} pipes context to {string}', function (this: CustomWorld, context
 
 When('messaging receives {string}', function (this: CustomWorld, field: string) {
   const message = handleResolve(field, this);
-  this.log(`Sending: ${JSON.stringify(message)}`);
-  this.messaging!!.receive(message, this.log);
+  //TODO: restore this
+  //this.log(`Sending: ${JSON.stringify(message)}`);
+  console.log(`Sending: `, message);
+
+  this.messaging!.receive(message, console.log); //this.log);
 });
 
 Then('messaging will have posts', function (this: CustomWorld, dt: DataTable) {
   // just take the last few posts and match those
   const matching = dt.rows().length;
-  var toUse = this.messaging?.allPosts!;
+  let toUse = this.messaging!.allPosts!;
```

Comment:
Are the `!` here necessary?

Reply:
Apparently so... The reference is set up in CustomWorld:

https://github.com/finos/FDC3/blob/7b330ac75f627b3c1faddba7f753547b08792b0c/packages/fdc3-agent-proxy/test/world/index.ts#L6

Then populated by a step that is used in the background for each test file:


https://github.com/finos/FDC3/blob/7b330ac75f627b3c1faddba7f753547b08792b0c/packages/fdc3-agent-proxy/test/step-definitions/generic.steps.ts#L15-L19)

If you can figure out something better I'll happily drop the !